src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -30,8 +34,27 @@
 
 public class JvmMonitor extends AbstractMonitor
 {
-  private final KeyedDiff gcDiff = new KeyedDiff();
-
+  private final Sigar sigar = SigarUtil.getSigar();
+  /**
+   * The following code is partially based on
+   * https://github.com/aragozin/jvm-tools/blob/e0e37692648951440aa1a4ea5046261cb360df70/
+   * sjk-core/src/main/java/org/gridkit/jvmtool/PerfCounterGcCpuUsageMonitor.java
+   *
+   * Names of counters could also be seen at
+   * http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/be698ac28848/
+   * src/share/classes/sun/tools/jstat/resources/jstat_options
+   */
+  private final JStatData jStatData = JStatData.connect(sigar.getPid());
```

Comment:
@logarithm It might have some considerable cost itself, but it doesn't affect JVM in a long run. It doesn't instrument all the code like tracing profilers do, doesn't add additional pauses to JVM, etc.